web/src/components/App.jsx

```diff
@@ -39,37 +41,47 @@ const darkModeEnabled = (prefersDarkMode, themePreference) => {
 };
 
 const App = () => {
+  const { i18n } = useTranslation();
+  const languageDir = i18n.dir();
+
   const [account, setAccount] = useState(null);
   const accountMemo = useMemo(() => ({ account, setAccount }), [account, setAccount]);
   const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
   const themePreference = useLiveQuery(() => prefs.theme());
   const theme = React.useMemo(
-    () => createTheme(darkModeEnabled(prefersDarkMode, themePreference) ? darkTheme : lightTheme),
-    [prefersDarkMode, themePreference]
+    () => createTheme({ ...(darkModeEnabled(prefersDarkMode, themePreference) ? darkTheme : lightTheme), direction: languageDir }),
+    [prefersDarkMode, themePreference, languageDir]
   );
 
+  useEffect(() => {
+    document.documentElement.setAttribute("lang", i18n.language);
+    document.dir = languageDir;
+  }, [i18n.language, languageDir]);
```

Comment:
Dumb question: 
- Should this not be `i18n.resolvedLanguage` everywhere? I had to do that in the preferences to avoid issues.
- Have you tested this with a cleared localStorage, i.e. without a language?

Reply:
- `resolvedLanguage` is specifically for preferences - it references a value in your supported languages, so you can match it to an option with the user. `language` is fine for all other use cases.

Reply:
- `resolvedLanguage` is specifically for preferences - it references a value in your supported languages, so you can match it to an option with the user. `language` is fine for all other use cases: https://www.i18next.com/overview/api#language
- works fine in incognito. it will use the fallback language if it can't detect one based on what the browser reports, or if it doesn't match a supported language.